src/ggsn_gn.erl

```diff
@@ -331,14 +331,16 @@ handle_request(ReqKey,
 
     gtp_context:terminate_colliding_context(ContextPreAuth),
 
-    SessionOpts0 = init_session(IEs, ContextPreAuth, AAAopts),
-    SessionOpts1 = init_session_from_gtp_req(IEs, AAAopts, SessionOpts0),
-    SessionOpts = init_session_qos(IEs, SessionOpts1),
-
     ergw_sx_node:wait_connect(SxConnectId),
     {ok, PendingPCtx, NodeCaps} = ergw_sx_node:select_sx_node(Candidates, ContextPreAuth),
 
     {ContextVRF, APNOpts} = ergw_gsn_lib:select_vrf_and_pool(NodeCaps, ContextPreAuth),
+
+    SessionOpts0 = init_session(IEs, ContextPreAuth, AAAopts),
+    SessionOpts1 = init_session_from_gtp_req(IEs, AAAopts, SessionOpts0),
+    SessionOpts2 = init_session_qos(IEs, SessionOpts1),
+    SessionOpts  = init_session_pool(ContextVRF, SessionOpts2),
+
```

Comment:
don't move this around, having it before ergw_sx_node:wait_connect/1 makes it a bit more parallel